### DISCUSSION TITLE
Augmented Reality does not work on iOS

### DISCUSSION THREAD

**Discussion by AmitMY** [google/model-viewer#4887] [Q&A]
(related to #3548)

### Description

in Safari in iOS, I can see the "mixed reality" button, while wrapped in a native app, using capacitor, it does not show. Anyway, even in safari, clicking it does not do anything, and there is nothing in the console.

| Safari | WKWebView |
| --------- | --------- |
| ![IMG_AE61C6490159-1](https://github.com/google/model-viewer/assets/5757359/a072a5ae-e3c6-4b50-acff-6f85e7771e54) | ![IMG_EE03C8B53D10-1](https://github.com/google/model-viewer/assets/5757359/797a0e0d-a950-4551-81bf-c09135ca5477) |

```html
<model-viewer
  #modelViewer
  alt="3D Sign Language Avatar"
  camera-controls
  camera-orbit="0deg 90deg auto"
  camera-target="0m 2.8m 0m"
  field-of-view="90deg"
  interaction-prompt="none"
  loading="eager"
  preload
  ar
  ar-scale="fixed"
  xr-environment
  src="3d/character.glb"
  ios-src="3d/character.usdz"
  ar-modes="scene-viewer webxr quick-look"></model-viewer>
```

#### Live Demo
Reproduction available in https://sign.mt

1. write "test"
2. hit the view switcher on the bottom right <img width="55" alt="image" src="https://github.com/google/model-viewer/assets/5757359/e79c97d7-d148-451e-88d4-57b4a3295d07">
3. and change to the avatar <img width="121" alt="image" src="https://github.com/google/model-viewer/assets/5757359/2d3984f3-a609-4170-a42b-7c9f5a5f7204">

### Version
- model-viewer: v3.4.0

### Browser Affected
- [x] Safari

### OS
- [x] iOS

### AR
- [ ] WebXR
- [ ] SceneViewer
- [ ] QuickLook

**Comment by elalish** [comment]
In your site I don't see the `ios-src` that you have in your comment. How much filesize does your GLB have? When you click the AR button, does it disappear and then reappear? Sometimes Safari is strangely slow to process USDZs.

**Comment by AmitMY** [comment]
It only adds the `ios-src` when you open the site from iOS (emulator or real device), because of the way my caching system works for offline usage.

<img width="592" alt="image" src="https://github.com/google/model-viewer/assets/5757359/c190be0e-87c9-4f09-8438-a9c42a06f89b">

In order to debug, I open the site on my iPhone, and then using safari developer tools view the iPhone dev tools.

Clicking the AR button does not flicker, doesn't do anything.

The GLB is 27mb: https://firebasestorage.googleapis.com/v0/b/sign-mt-assets/o/3d%2Fcharacter.glb?alt=media
The USDZ is 19mb (converted using reality converter): https://firebasestorage.googleapis.com/v0/b/sign-mt-assets/o/3d%2Fcharacter.usdz?alt=media

**Comment by elalish** [comment]
27mb is huge - you really shouldn't need more than 5 if you've properly optimized for mobile. You may be running out of GPU RAM. Can you make a simple repro with a smaller model and no fancy caching? I would guess the problem is caused by one of those two.

**Comment by omchiii** [comment]
I dont think that is the issue @elalish it just doesn't work on WKWebview
